credentials/insecure/insecure.go

```diff
@@ -71,6 +71,10 @@ func (info) AuthType() string {
 	return "insecure"
 }
 
+func (info) ValidateAuthority(string) error {
```

Comment:
A comment for this method please, mentioning that insecure credentials allows override the HTTP/2 `:authority` header to any value.